src/test/java/io/event/thinking/eventstore/sample/Subscription.java

```diff
@@ -0,0 +1,61 @@
+package io.event.thinking.eventstore.sample;
+
+import reactor.core.publisher.Mono;
+
+class Subscription {
```

Comment:
Here's the thing: To unsubscribe, you only need to check if the student and the course exist, and if the student is currently enrolled in that course. You don’t need to worry about course capacity when unsubscribing—that only matters during the subscription process. What I find beautiful about this approach is the flexibility to create as many models as necessary to validate commands, even having one per command if needed, instead of dealing with those large, complex aggregates that I’m more accustomed to.

Reply:
Actually, it's even simpler. We could do this with the criterion containing both tags, for course, and student. I'll build this.